hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1137,7 +1138,7 @@ func (r *HostedClusterReconciler) reconcileCAPIManager(ctx context.Context, crea
 // reconcileCAPIProvider orchestrates reconciliation of the CAPI provider
 // components for a given platform.
 func (r *HostedClusterReconciler) reconcileCAPIProvider(ctx context.Context, createOrUpdate upsert.CreateOrUpdateFN, hcluster *hyperv1.HostedCluster,
-	capiProviderDeploymentSpec *appsv1.DeploymentSpec) error {
+	capiProviderDeploymentSpec *appsv1.DeploymentSpec, p platform.Platform) error {
```

Comment:
consider p to be a pointer to avoid copying by value?

Reply:
Changed the assigned structures to be pointers

Reply:
I don't see the change, I'm still seeing you pass `p platform.Platform)` here. That's ok, this is not a blocker to me anyways.